File: snowball_primes.py
import math
 
def is_prime(n: int) -> bool:
    for i in range(2, int(math.sqrt(n)) + 1):
        if (n % i == 0):
            return False
    return True

if __name__ == "__main__":
    try:
        snowball = int(input("Enter the first digit of the snowball: "))
        #if not is_prime(snowball): raise ValueError

        snowballs = [snowball]

        while all([is_prime(snowball) for snowball in snowballs]):
            new_snowballs = []

            for snowball in snowballs:
                for i in range(1, 10, 2):
                    if is_prime(snowball*10+i):
                        new_snowballs.append(snowball*10+i)
            

            if not new_snowballs: break

            snowballs = new_snowballs

        
        print(snowballs[-1], "is the biggest snowball prime number from that starting point!")
            

    except ValueError:
        print("A prime integer is required. Try again.")

File: snowball_primes.go
package main

import (
	"fmt"
	"math"
)

func is_prime(n int) bool {
	for i := 2; i < int(math.Sqrt(float64(n)))+1; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	var snowball int
	fmt.Print("Enter the first digit of the snowball: ")
	fmt.Scanln(&snowball)

	snowballs := []int{snowball}

	for true {
		new_snowballs := []int{}

		for _, snowball := range snowballs {
			for i := 1; i < 10; i += 2 {
				if is_prime(snowball*10 + i) {
					new_snowballs = append(new_snowballs, snowball*10+i)
				}
			}
		}

		if len(new_snowballs) == 0 {
			break
		}

		snowballs = new_snowballs
	}

	fmt.Printf("%d is the biggest snowball prime number from that starting point!", snowballs[len(snowballs)-1])

}
